torch/lib/c10d/reducer.cpp

```diff
@@ -405,7 +405,7 @@ void Reducer::mark_variable_ready_dense(VariableIndex index) {
       // bucket_view. If gradient_as_bucket_view_ is set as true, let grad point
       // to bucket_view. If grad has already been set as views of buckets in
       // previous iterations, no copy is needed.
-      if (!grad.is_alias_of(bucket_view)) {
+      if (!c10d::shareSameStorage(grad, bucket_view)) {
```

Comment:
Can we add a unittest that would've failed if we used `is_alias_of` (presumably involving using ddp with gradient_as_bucket_view with same-storage tensors)?

Reply:
@rohan-varma it is not failing right now, but I think it is not properly used, see my replies to @albanD's comments.